Framework/Algorithms/src/GetEiMonDet2.cpp

```diff
@@ -425,7 +425,11 @@ void GetEiMonDet2::parseIndices(std::vector<size_t> &detectorIndices,
         spectraListParser.parse(getProperty(PropertyNames::DETECTORS)));
     auto back = std::back_inserter(detectorIndices);
     std::copy(detectors.begin(), detectors.end(), back);
-    monitorIndex = getProperty(PropertyNames::MONITOR);
+    int monitor = getProperty(PropertyNames::MONITOR);
+    if (monitor < 0) {
```

Comment:
Whichever way you prefer. If you decide to open a new issue please refer its number in this PR.